src/main/java/controller/LoginController.java

```diff
@@ -0,0 +1,57 @@
+package controller;
+
+import db.DataBase;
+import http.HttpQueryParams;
+import http.HttpRequest;
+import http.HttpResponse;
+import model.User;
+import session.Session;
+import session.SessionManager;
+
+import java.util.UUID;
+
+public class LoginController extends Controller {
+    private static final String INDEX_PAGE = "/index.html";
+    private static final String LOGIN_PAGE = "/user/login.html";
+    private static final String LOGIN_FAILED_PAGE = "/user/login_failed.html";
+
+    protected void doGet(HttpRequest request, HttpResponse response) {
+        String sessionId = request.getSessionId();
+
+        Session session = SessionManager.findSession(sessionId);
+
+        if (session != null) {
+            response.redirect(INDEX_PAGE);
+            return;
+        }
+        response.redirect(LOGIN_PAGE);
+    }
+
+    protected void doPost(HttpRequest request, HttpResponse response) {
+        String body = (String) request.getBody();
+        HttpQueryParams queryParams = HttpQueryParams.parseParams(body);
+        handleLogin(queryParams, response);
+    }
+
+    private void handleLogin(HttpQueryParams queryParams, HttpResponse response) {
+        String userId = queryParams.get("userId");
+        String password = queryParams.get("password");
+
+        User existUser = DataBase.findUserById(userId);
+
+        if (existUser == null || !existUser.match(userId, password)) {
+            response.redirect(LOGIN_FAILED_PAGE);
+            return;
+        }
+
+        String uuid = UUID.randomUUID().toString();
+        response.addCookie("JSESSIONID", uuid);
+        response.addCookie("logined", "true");
+        response.addCookie("Path", "/");
+
+        Session session = new Session(uuid);
```

Comment:
`httpRequest` 먼저 `session`을 가져와보고 없다면 새로운 세션 객체를 만들면 좋을 것 같아요. 🙂

Reply:
추가적으로 uuid를 외부에서 만들어서 주입해주는 것이 아니라 Session 객체 내부에서 직접 랜덤한 UUID를 생성해서 반환해주면 어떨까요? 

```suggestion
        Session session = new Session();
```


```java
    public Session() {
        this.id = UUID.randomUUID();
    }
```